File: api_wrapper/thecatapi/python/src/models/breed.py
import typing

from . import model


class Breed(model.Model):
    def __init__(self,
                 id: typing.Optional[int] = None,
                 name: typing.Optional[str] = None,
                 weight: typing.Optional[str] = None,
                 height: typing.Optional[str] = None,
                 life_span: typing.Optional[str] = None,
                 bred_for: typing.Optional[str] = None,
                 bred_group: typing.Optional[str] = None,
                 **other_fields) -> None:
        super().__init__(id, **other_fields)
        self.name = name
        self.weight = weight
        self.height = height
        self.life_span = life_span
        self.bred_for = bred_for
        self.bred_group = bred_group

File: api_wrapper/thecatapi/java/src/main/java/model/Breed.java
package model;

import java.util.Map;

public class Breed extends Model {
    private final String name;
    private final String weight;
    private final String height;
    private final String lifeSpan;
    private final String bredFor;
    private final String breedGroup;

    public Breed(int id, Map<String, Object> data) {
        super(id, data);
        this.name = String.valueOf(data.get("name"));
        this.weight = String.valueOf(data.get("weight"));
        this.height = String.valueOf(data.get("height"));
        this.lifeSpan = String.valueOf(data.get("life_span"));
        this.bredFor = String.valueOf(data.get("bred_for"));
        this.breedGroup = String.valueOf(data.get("breed_group"));
    }

    public String getName() {
        return name;
    }

    public String getWeight() {
        return weight;
    }

    public String getHeight() {
        return height;
    }

    public String getLifeSpan() {
        return lifeSpan;
    }

    public String getBredFor() {
        return bredFor;
    }

    public String getBreedGroup() {
        return breedGroup;
    }
}
